### PR TITLE
libopae++: add logging and exceptions

### PR DESCRIPTION
**Pull request by tswhison**
Adds logging and exceptions for the current error paths.

### CODE REVIEW

File: libopae++/src/dma_buffer.cpp [hunk 1]
```diff
@@ -38,15 +38,22 @@ dma_buffer::~dma_buffer() {
     fpga_result res = fpgaReleaseBuffer(handle_->get(), wsid_);
 
     if (res != FPGA_OK) {
-      // TODO log/throw error
+      log_.error() << "fpgaReleaseBuffer() failed with (" << res
+                   << ") " << fpgaErrStr(res);
```
Comment (r-rojo) [inline libopae++/src/dma_buffer.cpp]: doesn't except call fpgaErrStr?

File: libopae++/include/opaec++/pvalue.h [hunk 2]
```diff
@@ -66,29 +72,38 @@ struct guid_t {
...
+    fpga_result res;
+    if ((res = fpgaPropertiesGetGUID(props, &guid_value)) == FPGA_OK) {
       char guid_str[84];
       uuid_unparse(g.data_.data(), guid_str);
       ostr << guid_str;
     } else {
-      // TODO: Log or throw
+      g.log_.error() << "fpgaPropertiesGetGUID() failed with (" << res
+                     << ") " << fpgaErrStr(res);
+      throw except(res, OPAECXX_HERE);
```
Comment (r-rojo) [inline libopae++/include/opaec++/pvalue.h]: What if the `res` variable is `FPGA_NOT_FOUND`?

File: libopae++/include/opaec++/pvalue.h [hunk 3]
```diff
@@ -126,16 +141,20 @@ struct pvalue {
...
     fpga_properties props = *p.props_;
-    if (p.get_(props, &value) == FPGA_OK) {
+    fpga_result res;
+    if ((res = p.get_(props, &value)) == FPGA_OK) {
       ostr << +(value);
     } else {
-      // TODO: Log or throw
+      p.log_.error() << "property getter failed with (" << res
+                     << ") " << fpgaErrStr(res);
+      throw except(res, OPAECXX_HERE);
```
Comment (r-rojo) [inline libopae++/include/opaec++/pvalue.h]: What if the `res` variable is `FPGA_NOT_FOUND`?

File: libopae++/src/dma_buffer.cpp [hunk 4]
```diff
@@ -38,15 +38,22 @@ dma_buffer::~dma_buffer() {
...
 
 dma_buffer::ptr_t dma_buffer::allocate(handle::ptr_t handle, size_t len) {
   ptr_t p;
+  opae::fpga::internal::logger log("dma_buffer::allocate()");
 
-  if (!len) return p;
+  if (!len) {
+    log.error() << "attempt to allocate buffer with len == 0";
+    throw except(OPAECXX_HERE);
+    return p;
```
Comment (r-rojo) [inline libopae++/src/dma_buffer.cpp]: does the compiler complain if we don't return from here?

File: libopae++/include/opaec++/pvalue.h (hunk 2 above)
Comment (tswhison) [reply libopae++/include/opaec++/pvalue.h]: Ah, yes. Thanks

File: libopae++/include/opaec++/pvalue.h (hunk 3 above)
Comment (tswhison) [reply libopae++/include/opaec++/pvalue.h]: Yes, thanks.

**Pull request merged**